src/core/tests/library/RuntimeCompositor.py

```diff
@@ -41,7 +41,7 @@
 
 
 class RuntimeCompositor(object):
-    def __init__(self, argv=Constants.DEFAULT_UNSPECIFIED_VALUE, legacy_mode=False, package_manager_name=Constants.APT, vm_cloud_type=Constants.VMCloudType.AZURE):
+    def __init__(self, argv=Constants.DEFAULT_UNSPECIFIED_VALUE, legacy_mode=False, package_manager_name=Constants.APT, vm_cloud_type=Constants.VMCloudType.AZURE, test_type="HappyPath"):
```

Comment:
What is the need for this?

Reply:
The package manager initialization is done in this init function of RuntimeCompositor. The init function of YumPackageManager has function call get_yum_version which runs command "sudo yum --version". The sample output of "sudo yum --version" is added in the test_type "YumVersion4Dependency" for yum version 4. Without this change, the default test_type is "HappyPath" which does not contain output of "sudo yum --version". 

This code changes are done so that RuntimeCompositor can be initialized with input test_type. Initializing RuntimeCompositor with the given test_type will initialize LegacyEnvLayerExtensions with the given test_type and hence  package manager with sample output of given test_type. I have made the required changes in init function of LegacyEnvLayerExtensions also. Without this change, the package managers were always initialized with outputs in the test_type "HappyPath" as per init function of LegacyEnvLayerExtensions. Added parameter test_type in the init function of LegacyEnvLayerExtensions.

Reply:
For this test, i feel simply setting the yum_Version in your test as 4 will do. This seems to be too much change for this scenario.

Reply:
Had offline discussion. This will be need to test Code in the init methods of package manager,.